src/components/Collapsible/Collapsible.jsx

```diff
@@ -19,35 +19,33 @@ import classNames from 'classnames';
 import {propTypes, defaultProps} from './Collapsible.props';
 import './Collapsible.scss';
 
-export const Collapsible = ({expanded, children, duration, onTransitionEnd, ...props}) => {
+export const Collapsible = ({expanded, children, onTransitionEnd, ...props}) => {
     const [{motion, height}, setState] = useState({motion: '', height: expanded ? 'auto' : 0});
     const hasChildren = !!React.Children.count(children);
-    const timeout = useRef(), content = useRef();
+    const contentRef = useRef();
     const handleOnTransitionEnd = useCallback(e => {
-        if (e.propertyName === 'height') {
+        // 'transform' is the longest transition property out of multiple ones
+        if (e.propertyName === 'transform') {
+            setState(state => ({...state, motion: '', height: state.height ? 'auto' : '0'}));
             onTransitionEnd(e);
         }
     }, [onTransitionEnd]);
 
     useEffect(() => {
-        if (hasChildren) {
-            setState({height: content.current.clientHeight, motion: expanded ? 'expanding' : 'collapsing'});
-            clearTimeout(timeout.current);
-            timeout.current = setTimeout(() => {
-                // requestAnimationFrame is used here so that the state is set after the transition has been completed.
-                // Without it, the state may be set too early, and the onTransitionEnd will not be triggered for certain
-                // properties.
-                window.requestAnimationFrame(() => {
-                    setState(state => ({...state, height: expanded ? 'auto' : 0}));
-                });
-                // Remove the motion after the transition has been completed
-                timeout.current = setTimeout(() => {
-                    setState(state => ({...state, motion: ''}));
-                }, expanded ? 0 : duration);
-            }, expanded ? duration : 0);
-        }
-        return () => clearTimeout(timeout.current);
-    }, [hasChildren, expanded, duration]);
+        contentRef.current && setState({
+            height: contentRef.current.clientHeight, // if was "auto" - measure again and change
+            motion: expanded ? 'expanding' : 'collapsing',
```

Comment:
I noticed in the sandbox is that when expanding, we get both `.expanding` and `.expanded` during the transition, I think we should only have `.expanding` during the transition and `.expanded` once it's completed.

Reply:
Since you've made it this way, I thought it would be a breaking change to do that, so I did not touch this part. Do you want me to do it?

Reply:
Oh you're right it was. Still, I think we should change it. I doubt it will break anything.